packages/survey-core/src/question_matrixdropdownbase.ts

```diff
@@ -2458,24 +2458,26 @@ export class QuestionMatrixDropdownModelBase extends QuestionMatrixBaseModel<Mat
     }
   }
   public getFirstQuestionToFocus(withError: boolean): Question {
-    return this.getFirstCellQuestion(withError);
+    return this.getFirstCellQuestion(withError, true);
   }
   protected getFirstInputElementId(): string {
-    var question = this.getFirstCellQuestion(false);
+    var question = this.getFirstCellQuestion(false, true);
     return question ? question.inputId : super.getFirstInputElementId();
   }
   protected getFirstErrorInputElementId(): string {
-    var question = this.getFirstCellQuestion(true);
+    var question = this.getFirstCellQuestion(true, true);
     return question ? question.inputId : super.getFirstErrorInputElementId();
   }
-  protected getFirstCellQuestion(onError: boolean): Question {
+  protected getFirstCellQuestion(onError: boolean, generateRows?: boolean): Question {
+    if (generateRows) {
```

Comment:
Generate rows looks weird here: when we try to focus a question, it should be rendered already. Probably `this.generateVisibleRowsIfNeeded();` call should be in `getFirstQuestionToFocus` function.